Week-4/excersie1.js

```diff
@@ -0,0 +1,20 @@
+
+
+function displayLog(some) {
+    if(Number(some) % 5 == 0){
+        setTimeout(() => console.log("Reject the promise " + some),2000);
+    }else{
+        setTimeout(() => console.log("Resolve the promise " + some),1000);
+    }
+}
+
+function getNumber(mycallback){
+    let randNum = Math.floor((Math.random() * 100) + 1);
+    mycallback(randNum);
+}
+
+getNumber(displayLog);
```

Comment:
function getNumber(resolveTime, rejectTime){
    let randomNum = parseInt(Math.random()*100);
    console.log(randomNum);

    if (randomNum % 5 == 0){
        // Thenable object
        return {
            then(){
                setTimeout(() => console.log(`custom promise has been REJECTED after ${rejectTime}ms`), rejectTime)
            }
        }
    }
    return {
        // Thenable object
        then(){
            setTimeout(() => console.log(`custom promise has been RESOLVED after ${resolveTime}ms`), resolveTime)
        }
    }
}

getNumber(1000, 2000).then();

Reply:
Write code in Thenable form

Reply:
Thanks monu for reviewing. Will update the recommendation method.